src/pke/lib/schemebase/base-leveledshe.cpp

```diff
@@ -557,10 +564,17 @@ template <class Element>
 Ciphertext<Element> LeveledSHEBase<Element>::ComposedEvalMult(ConstCiphertext<Element> ciphertext1,
                                                               ConstCiphertext<Element> ciphertext2,
                                                               const EvalKey<Element> evalKey) const {
+    const auto cc                  = ciphertext1->GetCryptoContext();
     auto algo                      = ciphertext1->GetCryptoContext()->GetScheme();
+    const auto cryptoParams        = ciphertext1->GetCryptoParameters();
     Ciphertext<Element> ciphertext = EvalMult(ciphertext1, ciphertext2);
     algo->KeySwitchInPlace(ciphertext, evalKey);
-    ModReduceInPlace(ciphertext, BASE_NUM_LEVELS_TO_DROP);
+    uint32_t levelsToDrop = BASE_NUM_LEVELS_TO_DROP;
+    if (cc->getSchemeId() == CKKSRNS_SCHEME) {
```

Comment:
Why is this code here rather then in CKKS files?

Reply:
Fixed.

Reply:
This part was not changed. it could be replaced with the code from GetCompositeDegreeFromCtxt()

Reply:
We cannot use GetCompositeDegreeFromCtxt() since it is a protected member. In fact, we don't need this here. It can just resort to use the implementation from the override in LeveledSHERNS by inheritance. I will just fix it and remove this condition and related things around it. Thanks.

Reply:
i didn't mean the function at all, only "... the code ..." from the function body. then there is no change here